### DISCUSSION TITLE
some M1 builds stuck in preparing step

### DISCUSSION THREAD

**Discussion by dmelo** [codemagic-ci-cd/codemagic-docs#1574] [Q&A] [answered]
I'm not being able to run my builds on the latest minutes. I get not errors. It is just stack on `Preparing build machine`, until timeout. I'm using `mac_mini_m1`.

![Screenshot 2023-04-17 at 12-19-16 CI_CD for Android iOS Flutter and React Native projects](https://user-images.githubusercontent.com/89710/232534165-00dd3ed8-745f-4860-a8dd-7bd8a1fd350b.png)

**Comment by mikhail-tokarev** [comment]
hi @dmelo 
all services operate as expected. Can you share a build id if the issue still persists for you?

**Reply by dmelo** [reply]
@mikhail-tokarev . Here is a few build ids with that problem: `643d5fb12898effb2e9812c4`, `643d62e167b67f20b68d6caa`, `643d63e2ca09108cec73e1e2`, `643d6a6a46958a95b297a363` .

 I'm also trying mac_pro. With mac_pro it is building. So perhaps this is only affecting M1 build? Build `643d76041e4542abc497b0cf` .

**Reply by mikhail-tokarev** [reply]
one idea what came to my mind is trying to remove cached files and start a new build. You can do it in `App Settings > Caching`
Let me know if it helps

**Reply by dmelo** [reply]
Thanks @mikhail-tokarev . Just tried it on build `643d90f360778014ae4e2210`. But no luck. Still the same behavior, no logs and stuck on `Preparing build machine`.

**Reply by dmelo** [reply]
Is there a restriction regarding max number of secrets/env variables that I can register? I ask this because I think this was the last change I've made to my project on CM, before it started failing.

**Comment by tsimbalar** [comment]
We are seeing similar issues as well 

for instance : 
- https://codemagic.io/app/5fc6539af7698e06abe1becd/build/643e417058859d6896988866
- https://codemagic.io/app/5fc6539af7698e06abe1becd/build/643e417058859d6896988867
- https://codemagic.io/app/5fc6539af7698e06abe1becd/build/643e417058859d6896988868

with always the same symptoms 

![image](https://user-images.githubusercontent.com/160544/232703027-74f37e42-8b0a-4767-a0c3-5e3e356aa157.png)

**Reply by helinanever** [reply]
Hi @tsimbalar, your case is somewhat different and is due to a problem with downloading the cache. Can you try clearing the cache and see if this resolves the issue?

**Comment by mikhail-tokarev** [comment, marked as answer]
Thank you for reporting this! We have discovered an issue with yesterday's deployment and the fix is now live.

**Reply by tsimbalar** [reply]
I'm still having issues now, though 😓 

<img width="589" alt="image" src="https://user-images.githubusercontent.com/160544/232761832-f6bb8cbd-04c6-451c-845a-1c2a2734b9a5.png">

- https://codemagic.io/app/5fc6539af7698e06abe1becd/build/643e78d88c4d080e1fa9496c
- https://codemagic.io/app/5fc6539af7698e06abe1becd/build/643e78d88c4d080e1fa9496a
- https://codemagic.io/app/5fc6539af7698e06abe1becd/build/643e78d88c4d080e1fa9496b 
😓
